### ISSUE TITLE
SUM() macro produces NaNs when summing Eigen vectors

### ISSUE THREAD

**Issue by Expander** [FlexibleSUSY/FlexibleSUSY#132]
I compile FS with clang++ 7.0.1-8 (tags/RELEASE_701/final) on Debian Buster like so:
~~~
./configure --with-models= --with-optional-modules=test --with-cxx=clang++
make -j4
make test/test_sum.x
~~~
The following test fails with this configuration:
~~~
test/test_sum.x --run_test=test_sum_eigen_vector
~~~
The output is:
~~~
Running 1 test case...
test/test_sum.cpp(131): error: in "test_sum_eigen_vector": check v(0) == 2. has failed [nan != 2]
test/test_sum.cpp(132): error: in "test_sum_eigen_vector": check v(1) == 4. has failed [nan != 4]

*** 2 failures are detected in the test module "test_sum"
~~~
I.e. the SUM() macro produces an Eigen vector with NaNs.

**Comment by Expander** [comment]
The issue seems to be that `is_eigen_type<T>::value` yields true for `Acc`, but false for `Evaled`, although both types are identical:
~~~
Evaled = Eigen::Matrix<double, 2, 1, 0, 2, 1> = N5Eigen6MatrixIdLi2ELi1ELi0ELi2ELi1EEE
Acc = Eigen::Matrix<double, 2, 1, 0, 2, 1> = N5Eigen6MatrixIdLi2ELi1ELi0ELi2ELi1EEE
~~~
In particular, the following commands
~~~
std::cout << "is_eigen_type<Evaled> = " << is_eigen_type<Evaled>::value << '\n';
std::cout << "is_eigen_type<Acc> = " << is_eigen_type<Acc>::value << '\n';
~~~
yield:
~~~
Evaled = Eigen::Matrix<double, 2, 1, 0, 2, 1> = N5Eigen6MatrixIdLi2ELi1ELi0ELi2ELi1EEE
Acc = Eigen::Matrix<double, 2, 1, 0, 2, 1> = N5Eigen6MatrixIdLi2ELi1ELi0ELi2ELi1EEE
is_eigen_type<Evaled> = 0
is_eigen_type<Acc> = 1
~~~

**Comment by Expander** [comment]
@jhyeon Are you ok with fix 4cd31ea ?
